Extensions/ArtifactEngine/Providers/cacheProvider.ts

```diff
@@ -0,0 +1,103 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as readline from 'readline';
+import { resolve } from 'url';;
+var crypto = require('crypto');
+
+var tl = require('vsts-task-lib/task');
+
+import * as models from '../Models';
+import { ArtifactItemStore } from '../Store/artifactItemStore';
+import { Logger } from '../Engine/logger';
+
+export class CacheProvider implements models.IArtifactProvider {
+
+    public artifactItemStore: ArtifactItemStore;
+    
+    constructor(artifactCacheDirectory: string, artifactCacheKey: string, relPath?: string) {
+        this.artifactCacheDirectory = artifactCacheDirectory;
+        this.key = crypto.createHash('SHA256').update(artifactCacheKey).digest('hex');
+        this.relPath = relPath ? relPath : '';
+        this.cacheDirectory = path.join(artifactCacheDirectory, "ArtifactEngineCache", this.key);
+        this.cacheHashPath = path.join(this.cacheDirectory , 'artifact-metadata.csv');
+        this.makeOldHash()
+    }
+
+    getRootItems(): Promise<models.ArtifactItem[]> {
+        throw new Error("Not implemented");
+    }
+
+    getArtifactItems(artifactItem: models.ArtifactItem): Promise<models.ArtifactItem[]> {
+        throw new Error("Not implemented");
+    }
+
+    getArtifactItem(artifactItem: models.ArtifactItem): Promise<NodeJS.ReadableStream> {
+        var promise = new Promise<NodeJS.ReadableStream> ((resolve,reject) => {
+            if(artifactItem.fileHash && artifactItem.fileHash === this.oldFileHashMap[path.normalize(artifactItem.path.substring(this.relPath.length+1))]
+                && fs.existsSync(path.join(this.artifactCacheDirectory,"ArtifactEngineCache", this.key, path.normalize(artifactItem.path.substring(this.relPath.length+1))))) 
+            {
+                const inputStream = fs.createReadStream(path.join(this.artifactCacheDirectory,"ArtifactEngineCache", this.key, artifactItem.path.substring(this.relPath.length+1)));
+                inputStream.on('error',(err) => {
+                    reject(err);
+                })
+                resolve(inputStream);
+            }
+            else 
+                resolve(undefined);    
+        });
+        return promise;
+    }
+
+    public putArtifactItem(item: models.ArtifactItem, stream: NodeJS.ReadableStream): Promise<models.ArtifactItem> {
+        throw new Error("Not implemented");
+    }
+
+    public getRootLocation(): string {
+        throw new Error("Not implemented");
+    }
+
+    public getRootItemPath(): string {
+        throw new Error("Not implemented");
+    }
+
+    dispose(): void {
+    }
+    
+    public makeOldHash() : void  {
+        if(fs.existsSync(path.join(this.artifactCacheDirectory,"ArtifactEngineCache",this.key,"verify.json"))) {
```

Comment:
follow styling: there should be a space after , in arguments. Fix at other places too. Use vscode to format code.